Noh doc postprocessing sw

nilsolav opened the pull request:
Added a chapter on the use of the convention for typical post processing steps.

nilsolav committed:
Added a chapter on how the standard is intended to be used in post processing software
To be able to interchangeably use different software to process sonar data, there is a need for clarifying how the different steps in the standard can be used for the different processing steps.

nilsolav committed:
Misplaced the figure caption

nilsolav committed:
Postprocesssing steps and the convention
Added a suggested use of the convention for a few typical post processing steps.

gavinmacaulay reviewed (changes requested):
This commit changes every line in the document. Perhaps the line endings changed? Please adjust the commit so that we can see what lines actually changed.

gavinmacaulay commented on SONAR-netCDF4/docs/crr341.adoc:
```diff
@@ -96,6 +96,16 @@ This document and the convention that it describes will change over time to impl
 
+== The recommended use of the SONAR-netCDF4 convention in post processing software
+
+There are several processing steps that are common among the different postprocessing tools used by the community, and there is a growing need to use the different tools interchangeably. This chapter describes a few standard processing steps and outlines how the standard is intended to be used to facilitate the interchangeability. Note that the steps may be different between software solutions, a ...
```
```suggestion
There are several processing steps that are common among the different postprocessing tools used by the community, and there is a growing need to use the different tools interchangeably. This chapter describes a few standard processing steps and outlines how the standard is intended to be used to facilitate this interchangeability. Note that the steps may be different between tools, and that the convention only recommends how to store the data for the steps that are most common. We do not address how and in which order to process your data.
```

gavinmacaulay replied:
```suggestion
There are several processing steps that are common among the different postprocessing tools used by the community, and there is a growing need to use the different tools interchangeably. This chapter describes a few standard processing steps and outlines how the convention is intended to be used to facilitate the interchangeability. Note that the steps may be different between tools, and the convention only recommends how to store the data for the steps that are most common. We do not address how and in which order to process your data.
```

gavinmacaulay commented on SONAR-netCDF4/docs/crr341.adoc:
```diff
@@ -96,6 +96,16 @@ This document and the convention that it describes will change over time to impl
+
+There are several processing steps that are common among the different postprocessing tools used by the community, and there is a growing need to use the different tools interchangeably. This chapter describes a few standard processing steps and outlines how the standard is intended to be used to facilitate the interchangeability. Note that the steps may be different between software solutions, a ...
+
+Exchange of information between different parts in the processing chain is becoming increasingly important. This is particularly relevant for emerging deep learning and machine learning frameworks, that has hardware requirements that often are different than the computers that run the traditional desktop applications. Processing tools to be run on autonomous platforms are also a need, which often ...
```
```suggestion
Exchange of information between different parts in the processing chain is becoming increasingly important. This is particularly relevant for emerging deep learning and machine learning frameworks, that have hardware requirements that are often different to the computers that are used for traditional desktop applications. Processing tools that run on autonomous platforms are also a need, which often require a different suite of software solutions, typically tailored to low power situations. 
```

gavinmacaulay commented on SONAR-netCDF4/docs/crr341.adoc:
```diff
@@ -110,6 +120,10 @@ Each SONAR-netCDF4 file is intended to store data from one sonar mounted on one
 
 A design principle of SONAR-netCDF4 has been to focus on describing the acoustic backscatter data, not the overall purpose and context of why the data were collected. Such broader metadata are better stored in separate metadata systems and schema (e.g. citenp:[iso2014,ices2016]).
 
+Figure 1. Typical steps in a postprocessing chaing for acoustical water column data. (TODO: Link these boxes to the groups in the convention and check if there are other needs)
```
```suggestion
.Typical steps in a postprocessing chain for acoustical water column data. (TODO: Link these boxes to the groups in the convention and check if there are other needs)
[[postprocessing_data_flow_figure]]
image::postprocessingDataFlowFigure.svg[align="center"]

```

gavinmacaulay commented on SONAR-netCDF4/docs/crr341.adoc:
```diff
@@ -96,6 +96,16 @@ This document and the convention that it describes will change over time to impl
+
+Exchange of information between different parts in the processing chain is becoming increasingly important. This is particularly relevant for emerging deep learning and machine learning frameworks, that has hardware requirements that often are different than the computers that run the traditional desktop applications. Processing tools to be run on autonomous platforms are also a need, which often ...
+
+Typical steps in the postprocessing chain of marine sonar data consist of preprocessing and noise removal, bottom detection, classification, and integration (Figure 1). Here we outline the different groups in the convention that can support the input and output from these steps. If a processing software step can read the data from the standard, process the information and write back to the data u ...
```
```suggestion
Typical steps in the postprocessing chain of marine sonar data consist of preprocessing and noise removal, bottom detection, classification, and integration (<<postprocessing_data_flow_figure>>). Here we outline the different groups in the convention that can support the input and output from these steps. If a processing software step can read the data, process the information and write back to the data using the convention, other software can pick it up and process it further. 
```

nilsolav committed:
Minor change to the ML part

nilsolav committed:
Changed back to original charcater encoding and added the chapter.
It seems like asciidocFx changed the encoding. Not sure why. Back to good ol' Emacs.

nilsolav committed:
Updated encoding (again)
reverted back to a version that had the same encoding and added the changes using asciidocFX. It seems like emacs caused the problems. Not sure why since I used UTF8 in all cases. anyway, let's see if this works.

nilsolav committed:
Apply suggestions from code review

Accepted and added editorial suggestions from Gavin.

Co-Authored-By: Gavin Macaulay <gavin@macaulay.co.nz>

nilsolav committed:
Added the postprocessing_data_flow_figure

gavinmacaulay commented:
Looks fine. Some polishing probably needed, but this can happen in prep for a new release of the convention.

nilsolav commented:
Yes, let's do the language editing later. What is missing is the link to the groups later in the document.
